src/tools/cephfs/cephfs-meta-injection.cc

```diff
@@ -15,87 +17,88 @@ using std::string;
-         "\tconv : convert decimal ino to hex\n"  \
-         "\tlistc : list all obj of dir\n" \
-         "\tshowm : show the info of ino\n" \
-         "\tshowfn : show the fnode of dir\n" \
-         "\tamend : amend part of the meta data\n" \
-         "\tamendfn : amend fnode from file\n"
-         );
+  string mode;
+  po::options_description modeoptions("mode options");
+  modeoptions.add_options()
+      ("mode", po::value<string>(&mode),
+       "\tconv : convert decimal ino to hex\n"  \
+       "\tlistc : list all obj of dir\n"        \
+       "\tshowm : show the info of ino\n"          \
+       "\tshowfn : show the fnode of dir\n"        \
+       "\tamend : amend part of the meta data\n"   \
+       "\tamendfn : amend fnode from file\n"
+       );
 
-    po::positional_options_description p;
-    p.add("mode", 1);
-    
-    po::options_description all("all options");
-    all.add(modeoptions).add(general);
-    po::variables_map vm;
-    try{
-        po::store(po::command_line_parser(argc, argv).options(all).positional(p).allow_unregistered().run(), vm);
-    }catch(exception &e){
-        cerr << "error : " << e.what() << std::endl;
-        return -1;
-    }
-    catch(...){
-        cout << "param error" << std::endl;
-        return 0;
-    }
-    
-    boost::program_options::notify(vm);
-    if (vm.count("help")){
-        std::cout << version << std::endl;
-        std::cout << "usage : \n"
-                  << "  cephfs-meta-injection <conv|listc|showm|showfn|amend|amendfn> -r <fsname:rank> -i <ino>"
-                  << std::endl;
-        std::cout << "example : \n"
-                  << "  amend info of inode(1099531628828)\n"
-                  << "    cephfs-meta-injection showm -r cephfs_a:0 -i 1099531628828 -o out\n"
-                  << "    alter file\n"
-                  << "    cephfs-meta-injection amend -r cephfs_a:0 -i 1099531628828 --in out --yes-i-really-mean-it"
-                  << std::endl;
-        std::cout << all << std::endl;
-        return 0;
-    }
-    
-    if (mode == "conv"){
-        MetaTool::conv2hexino(ino.c_str());
-        return 0;
-    }
-    
-    MetaTool mt(vm.count("debug"));
-    int rc = mt.init();
-    if (rc != 0) {
-        std::cerr << "error in initialization: " << cpp_strerror(rc) << std::endl;
-        return rc;
-    }
-    rc = mt.main(mode, rank_str, minfo, ino, out, in, vm.count("yes-i-really-really-mean-it"));
-    if (rc != 0) {
-        std::cerr << "error (" << cpp_strerror(rc) << ")" << std::endl;
-        return -1;
-    }
+  po::positional_options_description p;
+  p.add("mode", 1);
+
+  po::options_description all("all options");
+  all.add(modeoptions).add(general);
+  po::variables_map vm;
+  try {
+    po::store(po::command_line_parser(argc, argv).options(all).positional(p).allow_unregistered().run(), vm);
+  } catch(exception &e) {
+    cerr << "error : " << e.what() << std::endl;
+    return -1;
+  }
+  catch(...) {
```

Comment:
could you move this line up into the previous line?

Reply:
@tchaikov  updated it. thx